driver-sync/src/main/com/mongodb/client/http/MongoClients.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.http;
+
+public final class MongoClients {
+
+    private MongoClients() {
```

Comment:
why is the constructor private? Is there a name for such a pattern.

Reply:
It's an utility class (just namespace for getMongoClient) and so to prevent creation of object I had to do this as java creates a default public empty constructor.
This pattern is used by original MongoClients class as well